opengrok-tools/src/main/python/opengrok_tools/utils/restful.py

```diff
@@ -34,18 +34,27 @@
 
 
 def do_api_call(verb, uri, params=None, headers=None, data=None):
+    logger = logging.getLogger(__name__)
+
     handler = getattr(requests, verb.lower())
     if handler is None or not callable(handler):
         raise Exception('Unknown HTTP verb: {}'.format(verb))
 
-    return handler(
+    logger.debug("{} API call: {} with data '{}' and headers: {}".
+                 format(verb, uri, data, headers))
+    r = handler(
         uri,
         data=data,
         params=params,
         headers=headers,
         proxies=get_proxies(uri)
     )
 
+    if r is not None:
```

Comment:
thinking about it, i don't remember if this can occur with requests, but even if it could occur (`r is None`) then you should probably raise an exception here (like when handler is none)

Reply:
makes sense